src/shims/posix/linux/foreign_items.rs

```diff
@@ -182,7 +182,19 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                     id if id == sys_futex => {
                         futex(this, args, dest)?;
                     }
-                    id => throw_unsup_format!("Miri does not support syscall ID {}", id),
+                    id => {
+                        let error_msg = format!(
+                            "unsupported Miri functionality: syscall ID {} is not emulated",
+                            id
+                        );
+                        if this.eval_context_ref().machine.panic_on_unsupported {
+                            // message is slightly different here to make automated analysis easier
+                            this.start_panic(error_msg.as_ref(), StackPopUnwind::Skip)?;
+                            return Ok(false);
+                        } else {
+                            throw_unsup_format!("{}", error_msg);
```

Comment:
Should this be cleaned up to avoid the double-format, or is this fine?

Reply:
Looks like you already cleaned it up?

Reply:
Well, formatting still technically happens twice (once above the `if`, once for `throw_unsup_format`. I have no idea if this would make another allocation in this case or if we care.

Reply:
I don't think I care, for now.^^